proposals/nnnn-resumption-data-error-handling.md

```diff
@@ -0,0 +1,81 @@
+# Handle response from HMI during resumption data
+
+* Proposal: [SDL-NNNN](nnnn-resumption-data-error-handling.md)
+* Author: [Alexander Kutsan](https://github.com/LuxoftAKutsan)
+* Status: **Awaiting review**
+* Impacted Platforms: [Core]
+
+## Introduction
+The aim of this feature is to handle HMI error responses while restoring the data during resumption process.
+
+## Motivation
+
+During resumption some data may not be restored due to HMI error response to add persistent data request.
+
+SDL should take care of these possible issues and fail resume data that might not be restored. 
+
+## Proposed solution
+
+During registration with correct hash_ID SDL should send all required requests/notification for restoring persistent data.
+
+List of persistent data to restore : 
+
+ - AddCommand (Menu + VR)
+ - AddSubMenu
+ - CreateInteractionChoiceSet
+ - SetGlobalProperties
+ - SubscribeButton
+ - SubscribeVehicleData
+ - SubscribeWayPoints
+
+After successful responses are received for all HMI requests, SDL should send successful response to mobile application `RegisterAppInterfaceResponse(success=true,result_code=SUCCESS)` 
+and OnHashChangeNotification.
+
+If HMI responds with any kind of error or does not respond to any requests sent during resumption, SDL should revert already restored data with appropriate RPCs:
+ - DeleteCommand
+ - DeleteSubMenu
+ - DeleteInteractionChoiceSet
+ - ResetGlobalProperties
+ - UnsubscribeButton
+ - UnsubscribeVehicleData
+ - UnsubscribeWayPoints
+
+In the case some data, like a subscription, is already used by other applications, this means that the subscription is actual and SDL should not send unsubscribe requests to HMI. 
+
+
+After reverting persistent data SDL should response `RegisterAppInterfaceResponse(success=true,result_code=RESUME_FAILED)` to mobile application.
+
+If multiple applications trying to restore the same subscription SDL should send the only first subscription to HMI. 
```

Comment:
This also seems to keep getting missed.  Please update to "If multiple applications are trying to restore the same subscription, SDL should..."